vision_evaluation/evaluators.py

```diff
@@ -816,3 +817,233 @@ def __init__(self):
         super().__init__(metric='SPICE')
         self.predictions = []
         self.targets = []
+
+
+class MattingEvaluatorBase(Evaluator):
+    """
+    Base class for image matting evaluator
+    """
+    def __init__(self):
+        super(MattingEvaluatorBase, self).__init__()
+        self.predictions = []
+        self.targets = []
+        self.metric = None
+
+    def add_predictions(self, predictions, targets):
+        """ Adding predictions and ground truth of images for image matting task
+        Args:
+            predictions: list of image matting predictions, [matting1, matting2, ...], shape: (N, ), type: PIL image object or Numpy array
+            targets: list of image matting ground truth, [gt1, gt2, ...], shape: (N, ), type: PIL image object or Numpy array
+        """
+        self.targets += targets
+        self.predictions += predictions
+
+    def reset(self):
+        super(MattingEvaluatorBase, self).reset()
+        self.targets = []
+        self.predictions = []
+
+    def _convert2binary(self, mask, threshold=128):
```

Comment:
is this better than a binary mask? i.e. just (mask >= threshold)

Reply:
resolved